test/orange/orangeinp/RevolvedPolygon.test.cc

```diff
@@ -0,0 +1,285 @@
+                                                      "",
+                                                      "rp@0.0.0.cz",
+                                                      "",
+                                                      "rp@0.0.0,rp@0.0.ou"};
+
+    auto const& u = this->unit();
+    EXPECT_VEC_EQ(expected_surface_strings, surface_strings(u));
+    EXPECT_VEC_EQ(expected_volume_strings, volume_strings(u));
+    EXPECT_VEC_EQ(expected_md_strings, md_strings(u));
+}
+
+//---------------------------------------------------------------------------//
+/* Test two-subregion case consisting of a cone subtracted from a cylinder.
+ \verbatim
+    3 _
+       |
+    2 _|    __________
+ z     |   /          |
+    1 _|  /           |
+       | /            |
+    0 _|/_____________|__________
+       |    |    |    |    |    |
+       0    1    2    3    4    5
+                   r
+ \endverbatim
+ */
+TEST_F(RevolvedPolygonTest, two_subregion)
+{
+    VecReal2 polygon{{1, 2}, {0, 0}, {3, 0}, {3, 2}};
+
+    this->build_volume(RevolvedPolygon{"rp", std::move(polygon)});
+
+    static char const* const expected_surface_strings[] = {
+        "Plane: z=0", "Plane: z=2", "Cone z: t=0.5 at {0,0,0}", "Cyl z: r=3"};
+    static char const* const expected_volume_strings[]
+        = {"all(+0, -1, -3, !all(+0, -1, -2))"};
+
+    static char const* const expected_md_strings[] = {
+        "",
+        "",
+        "rp@0.0.0.mz,rp@0.0.1.mz",
+        "rp@0.0.0.pz,rp@0.0.1.pz",
+        "",
+        "rp@0.0.0.kz",
+        "",
+        "rp@0.0.0,rp@0.0.iu",
+        "rp@0.0.1.cz",
+        "",
+        "rp@0.0.1,rp@0.0.ou",
+        "rp@0.0.nui",
+        "rp@0.0.d",
+    };
+
+    auto const& u = this->unit();
+    EXPECT_VEC_EQ(expected_surface_strings, surface_strings(u));
+    EXPECT_VEC_EQ(expected_volume_strings, volume_strings(u));
+    EXPECT_VEC_EQ(expected_md_strings, md_strings(u));
+}
+
+//---------------------------------------------------------------------------//
+/* Test case with a single concave region.
+ \verbatim
+    3 _
+       |
+    2 _|    __________
+ z     |    \         |
+    1 _|    /         |
+       |  /           |
+    0 _|/_____________|__________
+       |    |    |    |    |    |
+       0    1    2    3    4    5
+                   r
+ \endverbatim
+ */
+TEST_F(RevolvedPolygonTest, two_levels)
+{
+    VecReal2 polygon{{1, 2}, {1.2, 1.5}, {0, 0}, {3, 0}, {3, 2}};
+
+    this->build_volume(RevolvedPolygon{"rp", std::move(polygon)});
+
+    static char const* const expected_surface_strings[]
+        = {"Plane: z=0",
+           "Plane: z=2",
+           "Cone z: t=0.5 at {0,0,0}",
+           "Cyl z: r=3",
+           "Plane: z=1.5",
+           "Cone z: t=0.8 at {0,0,0}",
+           "Cone z: t=0.4 at {0,0,4.5}"};
+    static char const* const expected_volume_strings[]
+        = {"all(+0, -1, -3, !all(+0, -1, -2), !all(!all(+0, -1, -2), "
+           "any(all(+0, -4, -5), all(-1, +4, -6))))"};
```

Comment:
Sometimes you need to manually check for weird string wrapping before committing...
```suggestion
           "Cone z: t=0.4 at {0,0,4.5}",};
    static char const* const expected_volume_strings[]
        = {R"(all(+0, -1, -3, !all(+0, -1, -2), !all(!all(+0, -1, -2), any(all(+0, -4, -5), all(-1, +4, -6)))))",};
```

Reply:
`pre-commit` reformats what you have. I unwrapped the two offending strings in this file, added trailing `,`, and then let pre-commit handle it. Anything different would require `clang-format off`.

Reply:
Yep, that's all I was suggesting!